### PR TITLE
feat(events): Include metrics context in SQS events

### PR DESCRIPTION
**Pull request by rfk**
This will allow us to tunnel `utm_*` metrics parameters through to basket.  Connects to https://github.com/mozilla/fxa/issues/148

### CODE REVIEW

**Comment by rfk** [comment]
@philbooth mind a bit of f? on this?  It's basically trying to make the metrics context bundle show up in SQS events to other services in the same way that it shows up in our own activity events.

**Comment by rfk** [comment]
(I'm seeing occasional timeouts in the testsuite that I haven't dug into yet, so not ready for final r?)

File: lib/log.js [hunk 1]
```diff
@@ -70,12 +72,14 @@ Lug.prototype.begin = function (op, request) {
...
 }
 
-Lug.prototype.event = function (name, data) {
+Lug.prototype.event = function (name, request, data) {
   var e = {
     event: name,
     data: unbuffer(data)
   }
-  process.stdout.write(JSON.stringify(e) + '\n')
+  data.metricsContext = metricsContext.add({},
```
Comment (philbooth) [inline lib/log.js]: I think this should be `e.data.metricsContext`.

`unbuffer` returns a clone of `data` unless you pass the `inplace` argument, so you're setting `metricsContext` on the source object here, not the target.

**Comment by philbooth** [comment]
Other than that assignment to `data.metricsContext`, this looks good to me. 👍

File: lib/log.js (hunk 1 above)
Comment (rfk) [inline lib/log.js]: Indeed, nice catch - although I'm slightly confused that the tests were passing locally without this change

Comment (rfk) [inline lib/log.js]: Actually it looks like `unbuffer` in this file refers to a local function that does modify the object in place by default.  It's still clearer to spell out the full name though, so I'll change it.

**Comment by rfk** [comment]
The bug with the tests, turned out to be because I was creating a log object then failing to `close()` its stats client.  Pushing with fixes, let's see if it passes in travis...

**Comment by rfk** [comment]
Tests pass, @philbooth could you please do a final r?

**Comment by philbooth** [comment]
👍 r+

**Pull request merged**